qiskit/visualization/latex.py

```diff
@@ -603,7 +603,10 @@ def _build_barrier(self, node, col):
                     first = last = index
             pos = self._wire_map[self._qubits[first]]
             self._latex[pos][col - 1] += " \\barrier[0em]{" + str(last - first) + "}"
-            self._latex[pos][col] = "\\qw"
+            if node.op.label is not None:
+                pos = indexes[0]
+                label = node.op.label.replace(" ", "\\,")
+                self._latex[pos][col] = "\\cds{0}{^{\\mathrm{%s}}}" % label
```

Comment:
Perhaps it would make more sense to wrap the label in `\text` instead of `\mathrm` - the latter still uses math-mode spacing/kerning rules (hence the `\,`), whereas `\text` swaps back to the full text font.  It depends on `amsmath` being available (I think), but that's part of the de facto LaTeX standard library.  Users would be able to have parts of the label rendered in math-mode by including the `$ ... $` manually.

That's just on the basis that I would expect "label" to be a text field, not an equation one.  If we feel like it's more of an "equation" field, then the current math-mode way makes sense.

Reply:
`\text` doesn't work on my TeX Live version of LaTex. I don't think we'd want to do another `usepackage` for `amsmath` just for this, so I'd suggest we leave the `\mathrm` as is.

Reply:
Yeah, that's fair enough.  I'm surprised that nothing is already loading `amsmath`, since it's about as common as `import numpy` in mathematical Python code, but fair enough.  If it's not already loaded, yeah, let's not add the extra dependency.